src/core/machine-id-setup.c

```diff
@@ -68,6 +68,11 @@ static int generate_machine_id(const char *root, sd_id128_t *ret) {
                                 log_info("Initializing machine ID from KVM UUID.");
                                 return 0;
                         }
+                        /* on POWER, it's exported here instead */
+                        if (id128_read("/sys/firmware/devicetree/base/vm,uuid", ID128_UUID, ret) >= 0) {
+                                log_info("Initializing machine ID from KVM UUID.");
```

Comment:
This should be differentiated from the previous call, maybe
`"Initializing machine ID from KVM UUID in /sys/class/dmi/id/product_uuid."`
`"Initializing machine ID from KVM UUID in /sys/firmware/devicetree/base/vm,uuid."`

Reply:
The message will only be printed once. And I didn't want to change it and has the same meaning irrespective of where it came from: `dmi` or `devicetree` paths. As it really is the KVM UUID that KVM has setup and passed through. And only one or the other path exist on any given architecture. Personally i would even `#ifdef` the code, but that's not the systemd source code style.

Reply:
OK.